inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -260,6 +260,8 @@ InferenceEngine::details::CNNLayerCreator::CNNLayerCreator(const std::shared_ptr
         // res->params = params;
 
         auto castedLayer = ::ngraph::as_type_ptr<::ngraph::op::v1::BinaryConvolution>(node);
+        IE_ASSERT(castedLayer) << " Operation " << node->description() << " with name "
```

Comment:
I think Node class has << operator. I think it is better to have a single notation for printing nodes rather that manually print description and name.

Reply:
Will recheck.

Reply:
Node << operator prints a lot of redundant information which contains input/output shapes, names of inputs and etc.
Example:
```
[ ERROR ] AssertionFailed: layer->get_output_partial_shape(i).is_static() nGraph v0::Concat Concat_17955(GatherIE_35975[0]:f32{?}, GatherIE_35978[0]:f32{?}, GatherIE_35981[0]:f32{?}, GatherIE_35984[0]:f32{?}, GatherIE_35987[0]:f32{?}, GatherIE_35990[0]:f32{?}, GatherIE_35993[0]:f32{?}, GatherIE_35996[0]:f32{?}, GatherIE_35999[0]:f32{?}, GatherIE_36002[0]:f32{?}, GatherIE_36005[0]:f32{?}, GatherIE_36008[0]:f32{?}, GatherIE_36011[0]:f32{?}, GatherIE_36014[0]:f32{?}, GatherIE_36017[0]:f32{?}, GatherIE_36020[0]:f32{?}, GatherIE_36023[0]:f32{?}, GatherIE_36026[0]:f32{?}, GatherIE_36029[0]:f32{?}, GatherIE_36032[0]:f32{?}, GatherIE_36035[0]:f32{?}, GatherIE_36038[0]:f32{?}, GatherIE_36041[0]:f32{?}, GatherIE_36044[0]:f32{?}, GatherIE_36047[0]:f32{?}, GatherIE_36050[0]:f32{?}, GatherIE_36053[0]:f32{?}, GatherIE_36056[0]:f32{?}, GatherIE_36059[0]:f32{?}, GatherIE_36062[0]:f32{?}, GatherIE_36065[0]:f32{?}, GatherIE_36068[0]:f32{?}, GatherIE_36071[0]:f32{?}, GatherIE_36074[0]:f32{?}, GatherIE_36077[0]:f32{?}, GatherIE_36080[0]:f32{?}, GatherIE_36083[0]:f32{?}, GatherIE_36086[0]:f32{?}, GatherIE_36089[0]:f32{?}, GatherIE_36092[0]:f32{?}, GatherIE_36095[0]:f32{?}, GatherIE_36098[0]:f32{?}, GatherIE_36101[0]:f32{?}, GatherIE_36104[0]:f32{?}, GatherIE_36107[0]:f32{?}, GatherIE_36110[0]:f32{?}, GatherIE_36113[0]:f32{?}, GatherIE_36116[0]:f32{?}, GatherIE_36119[0]:f32{?}, GatherIE_36122[0]:f32{?}, GatherIE_36125[0]:f32{?}, GatherIE_36128[0]:f32{?}, GatherIE_36131[0]:f32{?}, GatherIE_36134[0]:f32{?}, GatherIE_36137[0]:f32{?}, GatherIE_36140[0]:f32{?}, GatherIE_36143[0]:f32{?}, GatherIE_36146[0]:f32{?}, GatherIE_36149[0]:f32{?}, GatherIE_36152[0]:f32{?}, GatherIE_36155[0]:f32{?}, GatherIE_36158[0]:f32{?}, GatherIE_36161[0]:f32{?}, GatherIE_36164[0]:f32{?}, GatherIE_36167[0]:f32{?}, GatherIE_36170[0]:f32{?}, GatherIE_36173[0]:f32{?}, GatherIE_36176[0]:f32{?}, GatherIE_36179[0]:f32{?}, GatherIE_36182[0]:f32{?}, GatherIE_36185[0]:f32{?}, GatherIE_36188[0]:f32{?}, GatherIE_36191[0]:f32{?}, GatherIE_36194[0]:f32{?}, GatherIE_36197[0]:f32{?}, GatherIE_36200[0]:f32{?}, GatherIE_36203[0]:f32{?}, GatherIE_36206[0]:f32{?}, GatherIE_36209[0]:f32{?}, GatherIE_36212[0]:f32{?}) -> (f32{?}) operation cannot be converted to Concat layer with name: 6520 because output with index 0 contains dynamic shapes: {?}. Try to use CNNNetwork::reshape() method in order to specialize shapes before the conversion.
```

I think it is better to use current approach:
```
[ ERROR ] AssertionFailed: layer->get_output_partial_shape(i).is_static() nGraph Concat operation with name: 6520 cannot be converted to Concat layer with name: 6520 because output with index 0 contains dynamic shapes: {?}. Try to use CNNNetwork::reshape() method in order to specialize shapes before the conversion.
```